benchmark/solver/solver_common.hpp

```diff
@@ -628,6 +639,12 @@ void run_solver_benchmarks(std::shared_ptr<gko::Executor> exec,
                        })) {
                 continue;
             }
+            // annotate the test case
+            // This string needs to outlive `test_case_range` to make sure we
+            // don't use its const char* c_str() after it was freed.
+            auto test_case_str = system_generator.describe_config(test_case);
```

Comment:
should we just use `const std::string` to keep the string data in the profiler?

Reply:
It's a bit of a balance between allocation overhead and convenience - using `std::string` would mean that the caller needs to allocate storage, while `const char*` works efficiently with string literals `"foo"`, but may require allocation otherwise, e.g. dynamically constructed strings. We could probably improve that overhead, e.g. in `ProfilerHook::on_apply_started`, by providing a small buffer on the stack that handles all but the most extreme cases.

Reply:
as a compromise, how about we either add a `string_profiling_scope_guard` that also owns the corresponding string or return a struct containing both the string and the scope guard from `annotate`?

Reply:
I went for the latter solution, since this seems specific to the benchmark use case for now.